src/java/org/apache/poi/hssf/record/ObjRecord.java

```diff
@@ -171,24 +172,28 @@ public int getRecordSize() {
 	public int serialize(int offset, byte[] data) {
 		int recSize = getRecordSize();
 		int dataSize = recSize - 4;
-		LittleEndianByteArrayOutputStream out = new LittleEndianByteArrayOutputStream(data, offset, recSize); // NOSONAR
-
-		out.writeShort(sid);
-		out.writeShort(dataSize);
 
-		if (_uninterpretedData == null) {
-
-			for (int i = 0; i < subrecords.size(); i++) {
-				SubRecord record = subrecords.get(i);
-				record.serialize(out);
-			}
-			int expectedEndIx = offset+dataSize;
-			// padding
-			while (out.getWriteIndex() < expectedEndIx) {
-				out.writeByte(0);
+		try (LittleEndianByteArrayOutputStream out =
+				new LittleEndianByteArrayOutputStream(data, offset, recSize)) { // NOSONAR
+			out.writeShort(sid);
+			out.writeShort(dataSize);
+
+			if (_uninterpretedData == null) {
+
+				for (int i = 0; i < subrecords.size(); i++) {
+						SubRecord record = subrecords.get(i);
+						record.serialize(out);
+					}
+				int expectedEndIx = offset+dataSize;
+				// padding
+				while (out.getWriteIndex() < expectedEndIx) {
+					out.writeByte(0);
+				}
+			} else {
+				out.write(_uninterpretedData);
 			}
-		} else {
-			out.write(_uninterpretedData);
+		} catch (IOException ioe) {
+			// should never happen in practice
```

Comment:
ditto, better to throw an unchecked exception for cases where exceptions are not expected at all